dlt/common/json/__init__.py

```diff
@@ -49,6 +54,8 @@ def custom_encode(obj: Any) -> str:
         return dataclasses.asdict(obj)  # type: ignore
     elif isinstance(obj, Enum):
         return obj.value  # type: ignore[no-any-return]
+    elif isinstance(obj, PydanticAnyUrl):
+        return str(obj) # will punycode encode the host if required
```

Comment:
could also be:
```suggestion
        return obj.unicode_string() # will NOT punycode encode the host
```
which would be more in line with not touching the data inside DLT